### ISSUE TITLE
1080p Picam vs 480p usb cam

### ISSUE THREAD

**Issue by matt-desmarais** [jhiggason/yahboomg1tank#10]
Since my camera mount is messed up I am leaning towards using a pi cam. I followed the instructions [here](https://www.raspberrypi.com/documentation/computers/camera_software.html#building-rpicam-apps) and also installed the picamera2 python api. its a much better camera for $5-$10, it's easier to work with and frees up a usb port. A pan tilt mechanism with the existing servos could be 3d printed, what do you think? I printed a basic mount for now.
![PXL_20231223_020943921](https://github.com/jhiggason/yahboomg1tank/assets/3732462/d44ec45e-84dc-4e41-a60e-41319f603d0a)
![output](https://github.com/jhiggason/yahboomg1tank/assets/3732462/defb462b-6b08-46fc-baa1-5b15c7f4b9b6)

**Comment by matt-desmarais** [comment]
works with [Camera Ros](https://github.com/christianrauch/camera_ros) I went through the issues and got it working, its not documented much but it does work!
<img width="1023" alt="Screenshot 2023-12-23 at 7 33 50 PM" src="https://github.com/jhiggason/yahboomg1tank/assets/3732462/75517b13-6ed4-4a4b-a119-e9ab1d22b3e7">

**Comment by jhiggason** [comment]
I totally agree getting a better camera on there...but I also think we should try and use the one it comes with as well.  The kit cost + raspberry pi is already a bit cost prohibitive for some people so anything we can do to teach folks how to do CV stuff without spending extra money would be beneficial.  

I'll probably start with the cruddy one it comes with and then I have one of the new, low light raspberry pi cameras I will set up as well.  I think there might be some big benefits to keeping the resolution low when first starting CV projects- particularly Graphics processing power to deal with 1080p vs 480p video.  More thoughts on this later +)

**Comment by jhiggason** [comment]
Oops didnt mean to close this one

**Comment by matt-desmarais** [comment]
sounds good. I happened to find some replacement parts
https://www.thingiverse.com/thing:4606347
https://www.thingiverse.com/thing:5257502

**Comment by matt-desmarais** [comment]
I got the 480p camera put together. Maybe the picam can be the camera for driving and then the USB cam could be the object recognition/tracking etc camera?

**Comment by jhiggason** [comment]
Dang thats a good price.  I think I paid 70 for an amazon warehouse model.  Sounds like you got the servo thingy together but if your still missing pieces let me know and I can model up the replacement parts.

**Comment by jhiggason** [comment]
So just FYI- I'm going to enable discussions here then we can move these issues to there instead and leave this for problems with the ROS2 code and other things we need to fix =)
